File: ACM/GO/71A_Way Too Long Words.go
/*
 * @Author: KasperFan && fanwlx@foxmail.com
 * @Date: 2023-03-18 01:50:15
 * @LastEditTime: 2023-03-18 01:50:31
 * @FilePath: /ACM/GO/71A_Way Too Long Words.go
 * @Description: This file is created for learning Code.
 * Copyright (c) 2023 by KasperFan in WFU, All Rights Reserved.
 */
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var n int
	fmt.Scan(&n)
	words := make([]string, n)

	for i := 0; i < n; i++ {
		fmt.Scan(&words[i])
		if len(words[i]) > 10 {
			words[i] = string(words[i][0]) + strconv.Itoa(len(words[i])-2) + string(words[i][len(words[i])-1])
		}
	}

	for _, word := range words {
		fmt.Println(word)
	}
}

File: ACM/Python/CodeForces/71A_Way Too Long Words.py
'''
Author: KasperFan && fanwlx@foxmail.com
Date: 2023-03-18 01:40:58
LastEditTime: 2023-03-18 01:45:54
FilePath: /ACM/Python/CodeForces/71A_Way Too Long Words.py
Description: This file is created for learning Code.
Copyright (c) 2023 by KasperFan in WFU, All Rights Reserved. 
'''
n = int(input())
words = []

for i in range(n):
    words.append(input())
    if len(words[i]) > 10:
        words[i] = (
            f"{words[i][0]}{len(words[i])-2}{words[i][len(words[i])-1]}")

for i in words:
    print(i)
